### DISCUSSION TITLE
Eigen Matrix support!

### DISCUSSION THREAD

**Discussion by stefanaerospace** [alandefreitas/matplotplusplus#390] [Ideas]
I did a lot of work in python when it came to prototyping in college, and to me, python is useless without Numpy and Matplotlib. Now with matplotlib you can directly dump a numpy matrix into it and have it plot. Eigen seems to be the Numpy equivalent for C++, yet matplot++ doesn't seem to support eigen matrices (tried putting one in image()). Tried looking around for how one might go about this in the repo's issues and found this:
https://github.com/alandefreitas/matplotplusplus/issues/226

But it seems like a solution already exists within OpenCV, maybe it should be ported over into this code base?
 https://docs.opencv.org/3.4/d0/daf/group__core__eigen.html

Saw this, not sure if this is applicable: https://github.com/alandefreitas/matplotplusplus/discussions/140

**Comment by alandefreitas** [comment]
It's certainly possible. I think it would be best to come up with concepts that match Eigen and similar libraries, or span/mdspan... so they can be peer dependencies rather that direct dependencies.
